services/frontend/src/scenes/Profile/ProfileLayout.js

```diff
@@ -24,25 +26,36 @@ const sections = [
 
 const ProfileLayout = ({projects, profile, isLoading}) => (
     <ScrollContainer className="profile">
-        {
-            (!profile || isLoading) ? (
-                <CircularProgress className="loading-button-indicator" />
-            ) : (
-                <>
-                    <NavSidebar
-                        scrollSpySelectors={sections}
-                        containerClass={containerClass}
-                    />
-                    <ProfileContent
-                        projects={projects}
-                        profile={profile}
-                    />
-                </>
-            )
-        }
+        <LoadingValidator
+            dependencies={[profile]}
+            renderOnFailedLoad={InvalidProfile()}
```

Comment:
I was having troubles rendering the component inside of `LoadingValidator` is why.

That said, I took another look in the react docs and put together a better solution by using multiple "slots" instead of using `children` and the prop `renderOnFailedLoad`:
```
        <LoadingValidator
            ...   //more props here
            renderOnLoad={
                <>
                     <NavSidebar
                     ...  //more code here
                </>
            }
            renderOnFailedLoad={
                <InvalidProfile />
            }
        />
```
and then `LoadingValidator` looks like:
```
        return (emptyDependency(dependencies)) ? (
            renderOnFailedLoad
        ) : (
            renderOnLoad
        );
```